api/app/src/test/kotlin/packit/unit/service/PreAuthenticatedLoginServiceTest.kt

```diff
@@ -0,0 +1,47 @@
+package packit.unit.service
+
+import org.junit.jupiter.api.assertThrows
+import org.mockito.kotlin.doReturn
+import org.mockito.kotlin.mock
+import org.springframework.http.HttpStatus
+import packit.exceptions.PackitException
+import packit.model.User
+import packit.security.profile.UserPrincipal
+import packit.security.provider.JwtIssuer
+import packit.service.PreAuthenticatedLoginService
+import packit.service.UserService
+import kotlin.test.Test
+import kotlin.test.assertEquals
+
+class PreAuthenticatedLoginServiceTest
+{
+    @Test
+    fun `can save user and issue token`()
+    {
+        val mockUser = mock<User>()
+        val mockUserPrincipal = UserPrincipal("test.uesr", "Test User", mutableListOf(), mutableMapOf())
```

Comment:
This isn't actually a mock. Probably `mock<UserPrincipal>()` is preferable, since these constructor args aren't used by the test (as you can tell from the misspelling `test.uesr`).

Reply:
Ha, ok, I just remembered why I did it this way - because Mockito can't mock final classes, so you can't do `mock<UserPrincipal>`, and it won't let you cast to the required type either. I will rename the value at least.